public/scripts/report.js

```diff
@@ -1,24 +1,80 @@
 (function(){
 
-  // GET LOCATION
-  var locationButton = document.getElementById('getLocationButton');
-
-  function getLocation() {
-    if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition(showPosition);
-    } else {
-      // non-supprt error handling
-      console.log('Geolocation is not supported by this browser.');
+  // GET USER'S LOCATION
+  function getLocation(){
+
+    var locationButton = document.getElementById('getLocationButton');
+
+    function getLocation() {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(storePosition);
+      } else {
+        // non-supprt error handling
+        console.log('Geolocation is not supported by this browser.');
+      }
+    };
+    // store user's location in button attribute
+    function storePosition(position) {
+      var getLocationButton = document.getElementById('getLocationButton');
+
+      var coords = {
```

Comment:
probably no need for this to be an object - we're going to store it as a string (something like `"33.1232,54.123123"` - you might as well do this here